### PR TITLE
Add Boonex Dolphin Oauth

### PR DESCRIPTION
**Pull request by geekgonecrazy**
Source: https://github.com/boonex/rocket.chat/tree/master/packages/rocketchat-dolphin
Author: @AlexTr 

@AlexTr any objections to us merging this upstream?

### CODE REVIEW

**Comment by engelgabriel** [comment]
@geekgonecrazy can you fix the style?

**Comment by geekgonecrazy** [comment]
@engelgabriel fixed :+1:

**Comment by AlexTr** [comment]
@geekgonecrazy thank you

There are some more code to merge:
- `packages/rocketchat-ui-account/account/avatar/prompt.html` & `server/methods/getAvatarSuggestion.coffee` - to use Dolphin Avatar in Rocket.Chat
- `packages/rocketchat-ui-flextab/flex-tab/tabs/userInfo.html` - to display link to the profile in user info
- `server/lib/accounts.coffee` - to make Dolphin admin Rocket.Chat admin

**Comment by geekgonecrazy** [comment]
@alextr awesome I'll take a look and maybe do another PR.  I knew this piece was pretty self contained so was easy to bring upstream.

File: packages/rocketchat-dolphin/i18n/en.i18n.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "API_Dolphin_URL" : "Dolphin URL",
+  "Accounts_OAuth_Dolphin" : "Dolphin Login",
+  "Accounts_OAuth_Dolphin_id" : "Dolphin Key",
+  "Accounts_OAuth_Dolphin_secret" : "Dolphin Secret"
```
Comment (engelgabriel) [inline packages/rocketchat-dolphin/i18n/en.i18n.json]: We don't need the word `Dolphin` repeated all over. Delete this file and use generic keys for the settings: URL, Login, Key, Secret

**Comment by geekgonecrazy** [comment]
@engelgabriel Changed

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 2]
```diff
@@ -62,6 +62,7 @@
   "Accounts_OAuth_Custom_Token_Path": "Token Path",
   "Accounts_OAuth_Custom_Token_Sent_Via": "Token Sent Via",
   "Accounts_OAuth_Custom_Username_Field": "Username field",
+	"Accounts_OAuth_Dolphin_Url": "Dolphin URL",
```
Comment (engelgabriel) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Do we need the word `Dolphin` here? can't we just use URL?

Comment (geekgonecrazy) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Removed, also changed the setting name to match the pattern used instead of starting with API_

**Pull request merged**